### PR TITLE
Added more text to NCID section

### LINKED ISSUE #3534: NEW_CONNECTION_ID text is too restrictive on receivers
This is a small point, but it's pointlessly creating pain in our implementation and possibly some others. In Sec 19.15 (NEW_CONNECTION_ID frame):

`An endpoint that receives a NEW_CONNECTION_ID frame with a sequence number smaller than the Retire Prior To field of a previously received NEW_CONNECTION_ID frame MUST immediately send a corresponding RETIRE_CONNECTION_ID frame that retires the newly received connection ID.`

IMO this is poorly formulated. If the receiver has already sent an RCID frame for the sequence number and that RCID has been acked, it may have thrown away state for that sequence number and shouldn't have to resend a frame the peer has received. In particular, a buggy or malicious peer might send a very old sequence number, or the NCID receiver might have preemptively retired a sequence number it hadn't seen when it got the retire-prior-to field.

Simply appending 'if has not already done so' to the text above would do the trick I think. I'm happy to file a PR unless most people think this is a terrible idea.

### PR DESCRIPTION
**Pull request by martinduke**
Fixes #3534

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -5579,8 +5585,8 @@ value.
 An endpoint that receives a NEW_CONNECTION_ID frame with a sequence number
 smaller than the Retire Prior To field of a previously received
 NEW_CONNECTION_ID frame MUST immediately send a corresponding
-RETIRE_CONNECTION_ID frame that retires the newly received connection ID.
-
+RETIRE_CONNECTION_ID frame that retires the newly received connection ID,
+unless it has already done so for that sequence number.
```
Comment (erickinnear) [inline draft-ietf-quic-transport.md]: Was the concern about doing this multiple times, or was it about the `immediately` part? That still wasn't clear based on the discussion on the list.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -5571,6 +5571,12 @@ Retire Prior To field MUST be less than or equal to the Sequence Number field.
 Receiving a value greater than the Sequence Number MUST be treated as a
 connection error of type FRAME_ENCODING_ERROR.
 
+Endpoints that receive a NEW_CONNECTION_ID frame SHOULD NOT retire connection
```
Comment (erickinnear) [inline draft-ietf-quic-transport.md]: I honestly find this (and that it is normative language) more confusing than not: it's incredibly specific about the requirement, and that requirement is the exact inverse requirement of Retire Prior To -- I get that that's the point, but the phrasing leads me down a confusing path. 
I went to suggest an alternative for wording (and a new location away from the frame mechanics and as the second to last paragraph of `{{retiring-cids}}`), but I'm struggling to find anything here that doesn't seem to cause additional confusion. 

I'm not quite seeing how any reading of the Retire _PRIOR_ To text leads one to try and retire all of the CIDs that aren't _PRIOR_ to the value received. From the description of the issue you ran into, the problem was less around RPT at all and more than you tried to throw away all other CIDs when you got a NCID frame, which might be something worth mentioning but is not this text.

Text more to that effect would probably go best around the active_cid_limit discussion where we indicate that you should probably keep as many as you said you would be willing for the peer to send you.

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-quic-transport.md]: I would also lose the "immediately".  That's not a useful word to use in this context.

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -5571,6 +5571,12 @@ Retire Prior To field MUST be less than or equal to the Sequence Number field.
...
+IDs with a sequence number equal to or greater than the Retire Prior To field
+solely as a result of receiving the frame. If the peer is trying to maintain a
+stock of provided CIDs, this can result in an infinite loop of
+NEW_CONNECTION_ID and RETIRE_CONNECTION_ID frames.
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: I don't think that we need this paragraph.

**Comment by martinthomson** [comment]
If you drop the first paragraph, this is just an editorial fix as far as I'm concerned. 

That is, not sending a frame that was previously sent is not a change.  And - assuming that you dump it - "immediately" has no defined meaning anyway.

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (martinduke) [reply draft-ietf-quic-transport.md]: @erickinnear The multiple times part.
@martinthomson good suggestion; I will patch it after I make dinner.

File: draft-ietf-quic-transport.md (hunk 2 above)
Comment (martinduke) [reply draft-ietf-quic-transport.md]: I am happy to jettison this paragraph to get the other bit through. The point is to say "here is a stupid thing to do and don't do it". If there's some other wording, or some other place to put it that's better, I'm fine with that too.

**Comment by martinduke** [comment]
@martinthomson I'm happy to drop the first paragraph, and I'm glad to hear that it's editorial. Two people on our team took the existing text to mean that we had to send RCID after receiving NCID even though we'd previously retired that sequence number.

Update coming soon.

**Comment by martinduke** [comment]
@martinthomson Feel free to change to "editorial"

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -5578,9 +5578,9 @@ value.
 
 An endpoint that receives a NEW_CONNECTION_ID frame with a sequence number
 smaller than the Retire Prior To field of a previously received
-NEW_CONNECTION_ID frame MUST immediately send a corresponding
-RETIRE_CONNECTION_ID frame that retires the newly received connection ID.
-
+NEW_CONNECTION_ID frame MUST send a corresponding RETIRE_CONNECTION_ID frame
```
Comment (nibanks) [inline draft-ietf-quic-transport.md]: "promptly send"? Though that's almost the same as immediately.

Comment (martinthomson) [reply draft-ietf-quic-transport.md]: Yeah, this is fine as is.

**Pull request merged**